src/frontend/src/handler/drop_mv.rs

```diff
@@ -28,6 +29,16 @@ pub async fn handle_drop_mv(
 
     let catalog_reader = session.env().catalog_reader();
 
+    {
+        let reader = catalog_reader.read_guard();
+        let source = reader.get_source_by_name(session.database(), &schema_name, &table_name)?;
+        if source.source_type == SourceType::Source {
```

Comment:
Can we have same names for source and mview? Or this is for `materialized source`?

Reply:
> Can we have the same names for source and mview?

No, I've tested. But this check is to ensure a suitable error hint. Otherwise when it's created as a source, and a drop materialized view command will display error "Catalog error: table not found: s"

```sql
dev=> create source s with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_1_partition_topic.1', 'kafka.brokers' = '127.0.0.1:29092' ) row format json;
CREATE_SOURCE
dev=> drop materialized view s;
ERROR:  Catalog error: table not found: s
```

Reply:
For normal mviews, the source with the `table_name` does not exist, so that the `?` will throw the "not-found" error.

Reply:
Yep I've found this issue. Thanks for reminding.